File: template.go
package main

import (
	"fmt"
	"os"
)

func waitForEnter() {
	var tmp string
	fmt.Println("Press anything to continue: ")
	fmt.Scan(&tmp)
}
func createSshKeypair(username string) {
	fmt.Println("Run:")
	fmt.Printf("   ssh-keygen -t rsa -f ~/%v", username)
	fmt.Println()
	waitForEnter()
}
func gitCommit(username string) {
	fmt.Println("Copy ~/new_key.pub into the `user_keys` Git repository, then run:")
	fmt.Printf("    git commit %v", username)
	fmt.Println()
	fmt.Println("    git push")
	waitForEnter()
}
func waitForBuild() {
	build_url := "http://example.com/builds/user_keys"
	fmt.Printf("Wait for the build job at %v to finish", build_url)
	fmt.Println()
	waitForEnter()
}
func retrieveUserEmail(username string) string {
	dir_url := "http://example.com/directory"
	fmt.Printf("Go to %v", dir_url)
	fmt.Println()
	fmt.Printf("Find the email address for user `%v`", username)
	fmt.Println()
	var email string
	fmt.Println("Paste the email address and press enter: ")
	fmt.Scan(&email)
	return email
}
func sendPrivateKey(email string) {
	fmt.Println("Go to 1Password")
	fmt.Println("Paste the contents of ~/new_key into a new document")
	fmt.Printf("Share the document with %v", email)
	fmt.Println()
	waitForEnter()
}

func main() {
	username := os.Args[1]
	createSshKeypair(username)
	gitCommit(username)
	waitForBuild()
	email := retrieveUserEmail(username)
	sendPrivateKey(email)
	fmt.Println("Done.")
}

File: template.lua
#!/usr/bin/env lua

function waitForEnter()
  print("Press enter to continue: ")
  io.read()
end

function createSshKeypair(username)
  print("Run:")
  print(string.format("   ssh-keygen -t rsa -f ~/%s", username))
  waitForEnter()
end

function gitCommit(username)
  print("Copy ~/new_key.pub into the `user_keys` Git repository, then run:")
  print(string.format("    git commit %s", username))
  print("    git push")
  waitForEnter()
end

function waitForBuild()
  build_url = "http://example.com/builds/user_keys"
  print(string.format("Wait for the build job at %s to finish", build_url))
  waitForEnter()
end

function retrieveUserEmail(username)
  dir_url = "http://example.com/directory"
  print(string.format("Go to %s", dir_url))
  print(string.format("Find the email address for user `%s`", username))
  print("Paste the email address and press enter: ")
  local email = io.read()
  print()
  return email
end

function sendPrivateKey(email)
  print("Go to 1Password")
  print("Paste the contents of ~/new_key into a new document")
  print(string.format("Share the document with %s", email))
  waitForEnter()
end

username = arg[1]
createSshKeypair(username)
gitCommit(username)
waitForBuild()
email = retrieveUserEmail(username)
sendPrivateKey(email)
print("Done.")
